openedx/core/djangoapps/content/learning_sequences/api/processors/enrollment_track_partition_groups.py

```diff
@@ -0,0 +1,61 @@
+# lint-amnesty, pylint: disable=missing-module-docstring
+import logging
+
+from common.lib.xmodule.xmodule.partitions.enrollment_track_partition_generator import (
+    create_enrollment_track_partition_with_course_id
+)
+from common.lib.xmodule.xmodule.partitions.partitions import (
+    ENROLLMENT_TRACK_PARTITION_ID,
+)
+from common.lib.xmodule.xmodule.partitions.partitions_service import get_user_partition_groups
+
+from .base import OutlineProcessor
+
+log = logging.getLogger(__name__)
+
+
+class EnrollmentTrackPartitionGroupsOutlineProcessor(OutlineProcessor):
+    """
+    Processor responsible for applying all enrollment track user partition group to the outline.
+
+    Confining the processor to only EnrollmentTrack user partition is a significant limitation.
+    Nonetheless, it is a step towards the goal of supporting
+    all partition schemes in the future
+    """
+    def __init__(self, course_key, user, at_time):
+        super().__init__(course_key, user, at_time)
+        self.user_groups = []
+        self.enrollment_track_groups = {}
+
+    def load_data(self):
+        user_partition = create_enrollment_track_partition_with_course_id(self.course_key)
+        self.enrollment_track_groups = get_user_partition_groups(self.course_key, [user_partition], self.user)
+
+    def _should_block_be_removed(self, block):
+        """
+        The function to test whether the user is part of the group of which,
+        the block is restricting the content to.
+        """
+        groups = block.user_partition_groups.get(ENROLLMENT_TRACK_PARTITION_ID)
+        if not groups:
```

Comment:
Is `0` ever a valid value for a group ID? If it is, you might want to make this more specific, like `if groups is None:`

Reply:
No, `0` cannot be a valid value for a group ID. Group IDs are enrollment values given by https://github.com/edx/edx-platform/blob/master/lms/envs/common.py#L4327